scrapy/__init__.py

```diff
@@ -30,13 +26,24 @@
 # Scrapy and Twisted versions
 __version__ = (pkgutil.get_data(__package__, "VERSION") or b"").decode("ascii").strip()
 version_info = tuple(int(v) if v.isdigit() else v for v in __version__.split("."))
-twisted_version = (_txv.major, _txv.minor, _txv.micro)
 
 
-# Ignore noisy twisted deprecation warnings
-warnings.filterwarnings("ignore", category=DeprecationWarning, module="twisted")
+def __getattr__(name: str):
+    if name == "twisted_version":
+        import warnings
+
+        from twisted import version as _txv
+
+        from scrapy.exceptions import ScrapyDeprecationWarning
+
+        warnings.warn(
+            "The scrapy.twisted_version attribute is deprecated, use twisted.version instead",
+            ScrapyDeprecationWarning,
+        )
+        return _txv.major, _txv.minor, _txv.micro
+
+    raise AttributeError
```

Comment:
Wondering since this a module not a class if we should raise `ImportError` instead.

Reply:
Tried that, don't work. 😄